src/JanusFtl.cpp

```diff
@@ -332,7 +333,18 @@ json_t* JanusFtl::QuerySession(janus_plugin_session* handle)
 #pragma region Private methods
 Result<std::vector<std::byte>> JanusFtl::ftlServerRequestKey(ftl_channel_id_t channelId)
 {
-    return serviceConnection->GetHmacKey(channelId);
+    std::shared_ptr<ServiceConnection> connection = getServiceConnection(channelId);
+    return connection->GetHmacKey(channelId);
+}
+
+std::shared_ptr<ServiceConnection> JanusFtl::getServiceConnection(ftl_channel_id_t channelId) 
```

Comment:
Consider taking `const std::shared_lock<std::shared_mutex>& streamDataLock` as a parameter to ensure whoever is calling this function holds a lock on `orchestratorRelayChannels` so the value doesn't change underneath you.